### PR TITLE
tests: drivers: build_all: regulator: add test

### PR DESCRIPTION
**Pull request by gmarull**
Add a new _test_ that is used to build all regulator drivers. SoC regulators, such as Raspberry Pi Pico core supply are excluded as they are typically built as part of the platform.

### CODE REVIEW

**Review by carlocaione — DISMISSED** [review]

File: tests/drivers/build_all/regulator/i2c.dtsi [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+	regulators {
+		compatible = "nordic,npm6001-regulator";
+
+		BUCK0 {};
+		BUCK1 {};
+		BUCK2 {};
+		BUCK3 {};
+		LDO0 {};
+		LDO1 {};
```
Comment (fabiobaltieri) [inline tests/drivers/build_all/regulator/i2c.dtsi:32]: Sorry not related to this but... the uppercase here looks out of place, I could maybe understand it for LDO but buck is not an acronym, and it's not like we write stuff like SPI uppercase in the devicetree. Should these be lowercase for coherency with other devicetree stuff?

**Review by fabiobaltieri — DISMISSED** [review]

File: tests/drivers/build_all/regulator/i2c.dtsi (hunk 1 above)
Comment (gmarull) [reply tests/drivers/build_all/regulator/i2c.dtsi:32]: It's following Linux convention for regulator node names

Comment (fabiobaltieri) [reply tests/drivers/build_all/regulator/i2c.dtsi:32]: Uh, can't argue with LINUX.

**Review by aasinclair — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
@gmarull should there be an entry for `tests/drivers/build_all/regulator/` in the mtaintainers file?

File: tests/drivers/build_all/regulator/i2c.dtsi (hunk 1 above)
Comment (gmarull) [reply tests/drivers/build_all/regulator/i2c.dtsi:32]: image

**Review by fabiobaltieri — APPROVED** [review]

**Review by carlocaione — APPROVED** [review]

**Review by aasinclair — APPROVED** [review]

**Pull request merged**